diff --git a/gdal/port/cpl_port.h b/gdal/port/cpl_port.h
index d33d20b..25dead8 100644
--- a/gdal/port/cpl_port.h
+++ b/gdal/port/cpl_port.h
@@ -140,7 +140,7 @@
 #include <stdarg.h>
 #include <string.h>
 #include <ctype.h>
-#include <limits.h>
+#include <linux/limits.h>
 
 #include <time.h>
 
@@ -1085,7 +1085,7 @@ inline bool operator!= (const bool& one, const MSVCPedanticBool& other) { return
 #include <set>
 #include <string>
 #include <cstddef>
-#include <limits>
+#include <linux/limits>
 #include <sstream>
 #include <fstream>
 #include <algorithm>
