--- ./etc/bashrc.orig	2020-07-04 12:26:22.158039558 +0200
+++ ./etc/bashrc	2020-07-04 12:27:45.462962399 +0200
@@ -68,7 +68,7 @@
 # [WM_COMPILER] - Compiler:
 # = Gcc | Clang | Icc | Cray | Amd | Arm | Pgi | Fujitsu |
 #   Gcc<digits> | Clang<digits>
-export WM_COMPILER=Gcc
+export WM_COMPILER=Clang
 
 # [WM_PRECISION_OPTION] - Floating-point precision:
 # = DP | SP | SPDP
@@ -106,8 +106,8 @@
 # --
 projectDir="${BASH_SOURCE:-${ZSH_NAME:+$0}}";
 [ -n "$projectDir" ] && projectDir="$(\cd $(dirname $projectDir)/.. && \pwd -L)" ||\
-projectDir="$HOME/OpenFOAM/OpenFOAM-$WM_PROJECT_VERSION"
-# projectDir="/opt/openfoam/OpenFOAM-$WM_PROJECT_VERSION"
+# projectDir="$HOME/OpenFOAM/OpenFOAM-$WM_PROJECT_VERSION"
+projectDir="@TERMUX_PREFIX@/opt/openfoam/OpenFOAM-$WM_PROJECT_VERSION"
 # projectDir="/usr/local/OpenFOAM/OpenFOAM-$WM_PROJECT_VERSION"
 ################################################################################
 # Or optionally hard-coded (eg, with autoconfig)
@@ -152,7 +152,7 @@
 
 # Capture values of old directories to be cleaned from PATH, LD_LIBRARY_PATH
 foamOldDirs="$WM_PROJECT_DIR $WM_THIRD_PARTY_DIR \
-    $HOME/$WM_PROJECT/$USER $FOAM_USER_APPBIN $FOAM_USER_LIBBIN \
+    $HOME/$WM_PROJECT/termux $FOAM_USER_APPBIN $FOAM_USER_LIBBIN \
     $WM_PROJECT_SITE $FOAM_SITE_APPBIN $FOAM_SITE_LIBBIN \
     $FOAM_MODULE_APPBIN $FOAM_MODULE_LIBBIN"
 
@@ -160,7 +160,7 @@
 export WM_PROJECT_DIR="$projectDir"
 
 # [WM_PROJECT_USER_DIR] - Location of user files
-export WM_PROJECT_USER_DIR="$HOME/$WM_PROJECT/$USER-$WM_PROJECT_VERSION"
+export WM_PROJECT_USER_DIR="$HOME/$WM_PROJECT/termux-$WM_PROJECT_VERSION"
 
 # [WM_PROJECT_SITE] - Location of site-specific (group) files
 # Default (unset) implies WM_PROJECT_DIR/site
