--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/c	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,12 @@
+include $(GENERAL_RULES)/Clang/c
+
+cARCH       =
+
+# Compile option is non-mandatory, but must be non-empty
+sinclude $(DEFAULT_RULES)/c$(WM_COMPILE_OPTION)
+
+cFLAGS      = $(cARCH) $(GFLAGS) $(cWARN) $(cOPT) $(cDBUG) $(LIB_HEADER_DIRS) -fPIC
+
+ctoo        = $(WM_SCHEDULER) $(cc) $(cFLAGS) -c $< -o $@
+
+include $(GENERAL_RULES)/Clang/link-c
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/c++	2020-07-04 14:22:04.470616105 +0200
@@ -0,0 +1,14 @@
+include $(GENERAL_RULES)/Clang/c++
+
+c++ARCH     = -m32
+
+include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
+
+c++FLAGS    = $(c++ARCH) $(GFLAGS) $(c++WARN) $(c++OPT) $(c++DBUG) $(ptFLAGS) $(LIB_HEADER_DIRS) -fPIC
+
+Ctoo        = $(WM_SCHEDULER) $(CC) $(c++FLAGS) -c $< -o $@
+cxxtoo      = $(Ctoo)
+cctoo       = $(Ctoo)
+cpptoo      = $(Ctoo)
+
+include $(GENERAL_RULES)/Clang/link-c++
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/c++~	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,14 @@
+include $(GENERAL_RULES)/Clang/c++
+
+c++ARCH     = 
+
+include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
+
+c++FLAGS    = $(c++ARCH) $(GFLAGS) $(c++WARN) $(c++OPT) $(c++DBUG) $(ptFLAGS) $(LIB_HEADER_DIRS) -fPIC
+
+Ctoo        = $(WM_SCHEDULER) $(CC) $(c++FLAGS) -c $< -o $@
+cxxtoo      = $(Ctoo)
+cctoo       = $(Ctoo)
+cpptoo      = $(Ctoo)
+
+include $(GENERAL_RULES)/Clang/link-c++
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/c++Debug	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,2 @@
+c++DBUG     = -g -DFULLDEBUG
+c++OPT      = -O0
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/cDebug	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,2 @@
+cDBUG       = -g -DFULLDEBUG
+cOPT        = -O0
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/c++Opt	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,2 @@
+c++DBUG     =
+c++OPT      = -O3
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/cOpt	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,2 @@
+cDBUG       =
+cOPT        = -O3
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/c++Prof	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,2 @@
+c++DBUG    = -pg
+c++OPT     = -O2
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/cProf	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,2 @@
+cDBUG       = -pg
+cOPT        = -O2
--- /dev/null	2020-07-04 07:25:57.571395009 +0200
+++ ./wmake/rules/linuxARM7Clang/general	2020-07-04 14:21:40.625378168 +0200
@@ -0,0 +1,9 @@
+CPP        = cpp -traditional-cpp $(GFLAGS)
+
+PROJECT_LIBS = -l$(WM_PROJECT) -ldl
+
+include $(GENERAL_RULES)/standard
+include $(GENERAL_RULES)/Clang/openmp
+
+include $(DEFAULT_RULES)/c
+include $(DEFAULT_RULES)/c++
diff -u -r ../rules.orig/General/Clang/c ./wmake/rules/General/Clang/c
--- ../rules.orig/General/Clang/c	2020-07-04 14:14:17.914357166 +0200
+++ ./wmake/rules/General/Clang/c	2020-07-04 15:29:10.867438565 +0200
@@ -1,6 +1,6 @@
 SUFFIXES += .c
 
-cc          = clang
+cc          = @TERMUX_COMPILER_PREFIX@-clang
 
 cARCH       =
 cDBUG       =
diff -u -r ../rules.orig/General/Clang/c++ ./wmake/rules/General/Clang/c++
--- ../rules.orig/General/Clang/c++	2020-07-04 14:14:17.914357166 +0200
+++ ./wmake/rules/General/Clang/c++	2020-07-04 15:29:10.870773563 +0200
@@ -1,6 +1,6 @@
 SUFFIXES += .C .cc .cpp .cxx
 
-CC          = clang++ -std=c++11
+CC          = @TERMUX_COMPILER_PREFIX@-clang++ -std=c++11
 
 c++ARCH     =
 c++DBUG     =
diff -u -r ../rules.orig/General/general ./wmake/rules/General/general
--- ../rules.orig/General/general	2020-07-04 14:14:17.917692164 +0200
+++ ./wmake/rules/General/general	2020-07-04 19:22:30.699075897 +0200
@@ -1,17 +1,17 @@
 #-------------------------------*- makefile -*---------------------------------
 WM_VERSION = OPENFOAM=2006
 
-AR         = ar
-ARFLAGS    = cr
-RANLIB     = ranlib
-CPP        = cpp
-LD         = ld
+AR         = @TERMUX_COMPILER_PREFIX@-ar
+ARFLAGS    = @TERMUX_COMPILER_PREFIX@-cr
+RANLIB     = @TERMUX_COMPILER_PREFIX@-ranlib
+CPP        = @TERMUX_COMPILER_PREFIX@-cpp
+LD         = @TERMUX_COMPILER_PREFIX@-ld
 
 GFLAGS     = -D$(WM_VERSION) \
              -DWM_$(WM_PRECISION_OPTION) -DWM_LABEL_SIZE=$(WM_LABEL_SIZE)
 GINC       =
-GLIBS      = -lm
-GLIB_LIBS  =
+GLIBS      = -lm -fuse-ld=bfd -Wl,-rpath=@TERMUX_PREFIX@/lib,--enable-new-dtags,--as-needed,-z,relro,-z,now -lPstream -L$(WM_PROJECT_DIR)/platforms/$(WM_ARCH)$(WM_COMPILER)$(WM_PRECISION_OPTION)$(WM_LABEL_OPTION)$(WM_COMPILE_OPTION)/lib/openmpi-system
+GLIB_LIBS  = 
 
 
 ARCHITECTURE_RULES = $(WM_DIR)/rules/$(WM_ARCH)
--- ./wmake/rules/General/mpi-mpicc-openmpi~	2020-12-22 15:30:37.000000000 +0100
+++ ./wmake/rules/General/mpi-mpicc-openmpi	2020-12-25 17:26:23.789428498 +0100
@@ -1,8 +1,8 @@
 #------------------------------------------------------------------------------
-# Use openmpi mpicc to obtain settings - mpicc must be available
+# Use openmpi mpicc to obtain settings
 
 PFLAGS  = -DMPICH_SKIP_MPICXX -DOMPI_SKIP_MPICXX
-PINC    = $(shell mpicc --showme:compile)
-PLIBS   = $(shell mpicc --showme:link)
+PINC    = -I@TERMUX_PREFIX@/include
+PLIBS   = -L$(MPI_ARCH_PATH)/lib -Wl,-rpath=@TERMUX_PREFIX@/lib,--enable-new-dtags -lmpi
 
 #------------------------------------------------------------------------------

diff -u -r ../rules.orig/linux64Clang/c++ ./wmake/rules/linux64Clang/c++
--- ../rules.orig/linux64Clang/c++	2020-07-04 14:14:17.921027163 +0200
+++ ./wmake/rules/linux64Clang/c++	2020-07-04 15:29:10.700688649 +0200
@@ -1,6 +1,6 @@
 include $(GENERAL_RULES)/Clang/c++
 
-c++ARCH     = -m64 -pthread
+c++ARCH     = -m64
 
 include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
 
diff -u -r ../rules.orig/linuxARM64Clang/c++ ./wmake/rules/linuxARM64Clang/c++
--- ../rules.orig/linuxARM64Clang/c++	2020-07-04 14:14:17.924362161 +0200
+++ ./wmake/rules/linuxARM64Clang/c++	2020-07-04 15:29:10.700688649 +0200
@@ -1,6 +1,6 @@
 include $(GENERAL_RULES)/Clang/c++
 
-c++ARCH     = -pthread
+c++ARCH     = 
 
 include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
 
diff -u -r ../rules.orig/linuxClang/c++ ./wmake/rules/linuxClang/c++
--- ../rules.orig/linuxClang/c++	2020-07-04 14:14:17.924362161 +0200
+++ ./wmake/rules/linuxClang/c++	2020-07-04 15:29:10.700688649 +0200
@@ -1,6 +1,6 @@
 include $(GENERAL_RULES)/Clang/c++
 
-c++ARCH     = -m32 -pthread
+c++ARCH     = -m32
 
 include $(DEFAULT_RULES)/c++$(WM_COMPILE_OPTION)
 
